lib/node_modules/@stdlib/math/stats/anova1/test/test.js

```diff
@@ -0,0 +1,190 @@
+	var values;
+	var i;
+
+	values = [
+		[],
+		[5]
+	];
+
+	for ( i = 0; i < values.length; i++ ) {
+		t.throws( badValue( values[i] ), Error, 'throws an error when provided '+values[i] );
+	}
+	t.end();
+
+	function badValue( value ) {
+		return function badValue() {
+			anova1( value, ['frank', 'philipp'] );
+		};
+	}
+});
+
+// Check that factor is an array
+tape( 'the function throws an error if the `factor` argument is not an array', function test( t ) {
+	var values;
+	var i;
+
+	values = [
+		5,
+		true,
+		NaN,
+		null
+	];
+
+	for ( i = 0; i < values.length; i++ ) {
+		t.throws( badValue( values[i] ), Error, 'throws an error when provided '+values[i] );
+	}
+	t.end();
+
+	function badValue( value ) {
+		return function badValue() {
+			anova1( [1, 2, 3, 4, 5], value );
+		};
+	}
+});
+
+// Check if factor has length bigger than 2
+tape( 'the function throws an error if the `factor` argument has a length smaller than 2', function test( t ) {
+	var values;
+	var i;
+
+	values = [
+		[],
+		[14],
+		['frank']
+	];
+
+	for ( i = 0; i < values.length; i++ ) {
+		t.throws( badValue( values[i] ), Error, 'throws an error when provided '+values[i] );
+	}
+	t.end();
+
+	function badValue( value ) {
+		return function badValue() {
+			anova1( [1, 2, 3], value );
+		};
+	}
+});
+
+// Check that there are more than one unique elemt in factor
+tape( 'the function throws an error if the `factor` argument has less than 2 unique elements', function test( t ) {
+	var values;
+	var i;
+
+	values = [
+		['frank', 'frank', 'frank'],
+		['alex', 'alex', 'alex', 'alex']
+	];
+
+	for ( i = 0; i < values.length; i++ ) {
+		t.throws( badValue( values[i] ), Error, 'throws an error when provided '+values[i] );
+	}
+	t.end();
+
+	function badValue( value ) {
+		return function badValue() {
+			anova1( [1, 2, 3], value );
+		};
+	}
+});
+
+// Check that the arguments have the same length
+tape( 'the function throws an error if the arguments have unequal lengths', function test( t ) {
+	// Two unique arguments for x and factor
+	var valuesX;
+	var valuesY;
+	var i;
+
+	valuesX = [
+		[1, 2, 3, 4, 5, 6, 7, 8],
+		[5, 6, 7, 4],
+		[1, 2, 3]
+	];
+
+	valuesY = [
+		['frank', 'alex', 'frank', 'alex'],
+		['philipp', 'alex', 'nugent'],
+		['john', 'delta beta']
+	];
+
+	for ( i = 0; i < valuesX.length; i++ ) {
+		t.throws( badValue( valuesX[i], valuesY[i] ), Error, 'throws an error when provided '+valuesX[i]+valuesY[i] );
+	}
+	t.end();
+
+	function badValue( valueX, valueY ) {
+		return function badValue() {
+			anova1( valueX, valueY );
+		};
+	}
+});
+
+// Exact value tests from example
+tape( 'the function produces expected values on simple inputs', function test( t ) {
+	var x;
+	var f;
+	var out;
+
+	x = [3, 4, 5, 6, 2, 5, 10, 12, 8, 10];
+	f = ['frank', 'philipp', 'nugent', 'frank', 'philipp', 'nugent', 'frank', 'philipp', 'nugent', 'frank'];
+	out = anova1(x, f);
+
+	t.equal(out.treatment_df, 2);
```

Comment:
Where do these values come from? We should add a comment, e.g. `// Tested against R:`. Better yet would be to have a script to generate the R fixtures automatically. But we can add that later.

Reply:
These values came from R testing side by side, forgot to add the R testing fixtures. Will do at a later date.